tempesta_fw/http.c

```diff
@@ -3391,6 +3578,11 @@ tfw_http_resp_cache_cb(TfwHttpMsg *msg)
 	TfwHttpReq *req = resp->req;
 
 	TFW_DBG2("%s: req = %p, resp = %p\n", __func__, req, resp);
+
+	if (TFW_CONN_H2(req->conn)) {
+		tfw_h2_resp_adjust_fwd(resp);
+		return;
+	}
```

Comment:
The code below and  in `tfw_http_resp_fwd()` procedure is very different from what we need for HTTP/2 response sending: in case of HTTP/2 we do not need neither `TFW_HTTP_B_RESP_READY`/`TFW_HTTP_B_REQ_DROP` flags, nor `seq_queue`/`seq_lock` functionality and corresponding auxiliary `ret_queue` processing. So it seems that it's easier to leave  `tfw_http_resp_fwd()` as is for HTTP/1.1 and make separate procedure for HTTP/2 (with `tfw_http_adjust_resp()`).

Reply:
OK, makes sense, but is it possible to separate the logic to h2 and h1 functions and do here only if-else calling one of the functions, list like https://github.com/tempesta-tech/tempesta/pull/1271/files#r300829427 ? Just wan to have more consistent code for both of HTTP/1 and HTTP/2 protocols.

Reply:
Done.